# AlwaysValid

- `AlwaysValid()`

Validates any input as valid.

```php
v::alwaysValid()->isValid('whatever'); // true
```

## Categorization

- Booleans

## Changelog

Version | Description
--------|-------------
  0.5.0 | Created

***
See also:

- [AlwaysInvalid](AlwaysInvalid.md)
