# AlwaysInvalid

- `AlwaysInvalid()`

Validates any input as invalid.

```php
v::alwaysInvalid()->isValid('whatever'); // false
```

## Changelog

Version | Description
--------|-------------
  0.5.0 | Created

***
See also:

- [AlwaysValid](AlwaysValid.md)
