# AlwaysInvalid

- `AlwaysInvalid()`

Validates any input as invalid.

```php
v::alwaysInvalid()->isValid('whatever'); // false
```

## Categorization

- Booleans

## Changelog

Version | Description
--------|-------------
  0.5.0 | Created

***
See also:

- [AlwaysValid](AlwaysValid.md)
- [When](When.md)
